src/tools/rbd/action/Trash.cc

```diff
@@ -320,6 +345,148 @@ int execute_list(const po::variables_map &vm) {
   return 0;
 }
 
+void get_purge_arguments(po::options_description *positional,
+                            po::options_description *options) {
+  at::add_pool_options(positional, options);
+  at::add_no_progress_option(options);
+  
+  options->add_options()
+      (EXPIRED_BEFORE.c_str(), po::value<std::string>()->value_name("date"), 
+       "purges images that expired before the given date");
+  options->add_options()
+      (THRESHOLD.c_str(), po::value<double>(), 
+       "purges images until the current pool data usage is reduced to X%, "
+       "value range: 0.0-1.0");
+}
+
+
+int execute_purge (const po::variables_map &vm) {
+  size_t arg_index = 0;
+  std::string pool_name = utils::get_pool_name(vm, &arg_index);
+
+  librados::Rados rados;
+  librados::IoCtx io_ctx;
+  int r = utils::init(pool_name, &rados, &io_ctx);
+  if (r < 0) {
+    return r;
+  }
+
+  librbd::RBD rbd;
+  
+  std::vector<librbd::trash_image_info_t> trash_entries;
+  r = rbd.trash_list(io_ctx, trash_entries);
```

Comment:
I was thinking that ```purge``` should probably only handle images where the source == ```USER``` to avoid deleting RBD-internally created trash entries (i.e. rbd-mirror deferred deletions and/or migrating images).

Reply:
done